cinema/tests/test_movie_api.py

```diff
@@ -157,3 +163,126 @@ def test_image_url_is_shown_on_movie_session_detail(self):
+    def setUp(self) -> None:
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@user.com", password="13flrmkgse"
+        )
+        self.super_user = get_user_model().objects.create_superuser(
+            email="test@super_user.com", password="13flrmkgse"
+        )
+
+    def test_unauthenticated_user_get_request(self):
+        result = self.client.get(reverse("cinema:movie-list"))
+        self.assertEqual(result.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_authenticated_user_get_request(self):
+        self.client.force_authenticate(self.user)
+        result = self.client.get(reverse("cinema:movie-list"))
+        self.assertEqual(result.status_code, status.HTTP_200_OK)
+
+    def test_authenticated_user_post_request_forbidden(self):
+        self.client.force_authenticate(self.user)
+        result = self.client.post(reverse("cinema:movie-list"))
+        self.assertEqual(result.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_authenticated_superuser_post_request_allowed(self):
+        self.client.force_authenticate(self.super_user)
+        path = reverse("cinema:movie-list")
+        data = {
+            "title": "Sample movie",
+            "description": "Sample description",
+            "duration": 90,
+            "genres": [sample_genre().id],
+            "actors": [sample_actor().id],
+        }
+        result = self.client.post(path=path, data=data)
+        self.assertEqual(result.status_code, status.HTTP_201_CREATED)
+
+
+class MovieViewSetGetQuerySetTestCase(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@user.com", password="13flrmkgse"
+        )
+        self.client.force_authenticate(self.user)
+
+        self.top_gun = sample_movie(title="Top Gun")
+        self.snatch = sample_movie(title="Snatch")
+
+        self.action = sample_genre(name="Action")
+        self.comedy = sample_genre(name="Comedy")
+
+        self.tom_cruise = sample_actor(first_name="Tom", last_name="Cruise")
+        self.brad_pitt = sample_actor(first_name="Brad", last_name="Pitt")
+
+        self.top_gun.genres.add(self.action)
+        self.top_gun.actors.add(self.tom_cruise)
+
+        self.snatch.genres.add(self.comedy)
+        self.snatch.actors.add(self.brad_pitt)
+
+    def test_get_queryset_title_filter(self):
+        url = reverse("cinema:movie-list")
+        query_params = {"title": "Top Gun"}
+        response = self.client.get(url, query_params)
+        serializer_top_gun = MovieListSerializer(self.top_gun)
+        serializer_all_movie = MovieListSerializer(
+            Movie.objects.all(), many=True
+        )
+
+        self.assertIn(serializer_top_gun.data, response.data)
+        self.assertNotEqual(serializer_all_movie.data, serializer_top_gun)
+
+    def test_get_queryset_actors_filter(self):
```

Comment:
what about genre ?

Reply:
> what about genre ?

It is not a problem to add this test, I was afraid to have a problem with DRY in this case